Add setup and usage documentation to README (#53)
simple-icons/simple-icons-font: README.md

Free SVG icon font for popular brands. See them all on one page at <a href="https://simpleicons.org">SimpleIcons.org</a>. Contributions, corrections & requests can be made on GitHub. Started by <a href="https://twitter.com/bathtype">Dan Leech</a>.</p>
</p>

## Setup

### From CDN

```html
<link rel="stylesheet"

>       href="https://cdn.jsdelivr.net/npm/simple-icons-font@v2/font/simple-icons.min.css"

Reviewer: I think we should add the example for both JSDelivr and Unpkg, like in the main repo, agreed?

Reviewer: Also, wow this works I actually wasn't sure it would find the font files just like that :sweat_smile:

Author: > I think we should add the example for both JSDelivr and Unpkg, like in the main repo, agreed?

Yes!

> Also, wow this works I actually wasn't sure it would find the font files just like that sweat_smile

CSS magic :sparkles: It seems that relative calls in `url()` are relative to the URL of file calling them, which it's very nice.

      type="text/css">
```

> 
> ### NodeJS

Reviewer: I think we should add a note here about the fact that this limits the scope to `v2` like we do in the main repository.

```suggestion
These examples use the latest major version. This means you won't receive any updates following the next major release. You can use `@latest` instead to receive updates indefinitely. However, this will result in a icons not showing as expected if the icon is removed.

### NodeJS
```

Author: But only the NodeJS setup uses the latest major version with automatic updates not? Although the manual download gets the latest version, it's not updated automatically, so this warning should be inside "NodeJS" section :thinking:

Reviewer: This note is intended for the CDN users. The comment isn't even correct for NPM/NodeJS users, as installing `@latest` will not give you updates indefinitely, only until the next major release (same as just `npm install simple-icon-font`). We could add a similar note to the NodeJS section as well, though I would expect a bit more knowledge in this regard from NPM users (but maybe that is unfair).

The reason, I think, we should add it here is because more often than not when I see packages offer CDN guidelines  in their docs the URL doesn't include a version, so the fact that we do include it may be unexpected to users of CDNs.

Author: I see. I've included changing only "if the icon is removed" by "if an icon is removed".

Reviewer: Good catch, forgot that removing one icon may lead to problems with other icons as well :+1:

Unfortunately I made another mistake and right now the sentence still isn't correct:

```diff
- However, this will result in a icons not showing as expected if an icon is removed.
+ However, this will result in icons not showing as expected if an icon is removed.
```

Reviewer: Can you fix this as well @mondeja?

Author: Well seen, it's done :+1:

Reviewer: > Good catch, forgot that removing one icon may lead to problems with other icons as well 👍

Hmm actually, this isn't true right? If an icon is removed from the main project, then other icons should still show up correctly when using CSS classes even though the exact unicode value changed?

If so, then the sentence as it is currently written can be a bit misleading. Currently it can legitemately be interpreted as meaning: if any single icon is removed from the main project, then all icons in the font can get messed up. By the above reasoning this is not true, Therefore, I think we should write something like:

> However, this will result in icons not showing as expected when they are removed.

or

> However, this will result in icons not showing as expected if removed.

Author: What about this?

> However this may cause an icon to disappear if it has been removed in the latest version.

Reviewer: Also works :+1:

```
$ npm install simple-icons-font
```

After installation, the icon font can then be found in `node_modules/simple-icons-font/font` along with stylesheets for a basic setup.

### Manual download

1. [Download the latest version of the package][npm-registry-tarball-link], extract it, and move the content of the `font` folder to your desired location.

> 1. Reference the CSS file in a `link` tag of your HTML:

Author: OK. But I've used "Then, reference the CSS file" in the second to avoid repeat "You can" two times.

```html
<link rel="stylesheet"
      href="/path/to/simple-icons.min.css"
      type="text/css">
```

## Usage

The CSS files of simple-icons-font contains next classes:

- `simpleicons`: Adds the font to the container that will display the icon.
- `simpleicons--color`: Colorizes the icon with the main color of the brand.
- `simpleicons-[ICON SLUG]`: Replace here `[ICON SLUG]` by with the correspondent slug for the icon. Their slugs correpond to filenames, which you can see listed in [simple-icons/icons][simple-icons--icons-dir-link].

### Example

```html
<i class="simpleicons simpleicons--color simpleicons-simpleicons"></i>

> ```

Reviewer: I think an example will do to let people know how this works.

    Use any of the icons available in simple-icons by adding the following classes to a node in your HTML. Use the `simple-icons--color` class to apply the brand's color to the icon.

    ```html
    <i class="simpleicons simpleicons-[ICON NAME]"></i>
    <i class="simpleicons simpleicons-[ICON NAME] simple-icons--color"></i>
    ```

    Where `[ICON NAME]` is replaced by the icon name, for example:

    ```html
    <i class="simpleicons simpleicons-simpleicons"></i>
    <i class="simpleicons simpleicons-simpleicons simple-icons--color"></i>
    ```

    In this example we use the `<i>` tag, but any inline HTML tag should work as you expect.

We could still add more detailed documentation if it turns out to be necessary/you feel its necessary, but I would put it after the example. What do you think?

Also, I don't feel like it is necessary to explain that the `[ICON NAME]` should be the slug, at least we don't do it in the main project and it seems most people figure it out. That said, I'm not against adding it. Maybe wait and hope someone opens an issue if they experience difficulty because of this?

## Status

[![Build Status][build-status-image]][build-status-link]
[![NPM version][npm-version-image]][npm-package-link]

> [build-status-image]: https://img.shields.io/github/workflow/status/simple-icons/simple-icons-font/Verify/develop?logo=github

Reviewer: I guess using this badge is a bit nicer, mind pushing an update for this to `develop` for the main repo as well? (Or open a PR in case you think other maintainers should have a look at this first)

Author: I think we should avoid direct commits as much as possible to appear as transparent as possible. I'll upload a pull so someone could learn something about how to use shields.io, since they use simple-icons for icons.

Free SVG icon font for popular brands. See them all on one page at <a href="https://simpleicons.org">SimpleIcons.org</a>. Contributions, corrections & requests can be made on GitHub. Started by <a href="https://twitter.com/bathtype">Dan Leech</a>.</p>
</p>

## Setup

### From CDN

> ```html
> <html>
>   <head>
>     <meta charset="utf-8">
>     <link rel="stylesheet"
>           href="https://cdn.jsdelivr.net/npm/simple-icons-font/font/simple-icons.min.css"
>           type="text/css">
>   </head>
>   <body>
>     <i class="simpleicons simpleicons-simpleicons simpleicons--color"></i>
>   </body>
> </html>
> ```

Reviewer: I would prefer a simpler style similar to [the README of the main project](https://github.com/simple-icons/simple-icons#readme).

Also I think we should include the version here as well, as we do in the main project.

```suggestion
```html
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/simple-icons-font@v2/font/simple-icons.min.css" type="text/css">
```
```

> ### Manual download

Reviewer: I would say this is the least recommended way of using the font, can we list this one last?

> 1. Download the latest version of the package from
> [here][npm-registry-tarball-link], extract it and move the content of the
> `font` folder to your desired location. 

Reviewer: Could you keep the formatting style that is used in the main repository and write this as a single line? Also, using the text "here" for a link is bad for accessibility and often unnecessary, e.g here you can just do

```suggestion
1. [Download the latest version of the package][npm-registry-tarball-link], extract it, and move the content of the `font` folder to your desired location.
```

<sup>Alternatively, since this is technically invalid MarkDown, the two alternative lines should be indented with 3 spaces to line up properly.</sup>

1. Reference the CSS file in a `link` tag of your HTML:

```html
<html>
  <head>
    <meta charset="utf-8">

>     <link rel="stylesheet" href="simple-icons.min.css" type="text/css">

Reviewer: i would only leave this part of the sample code in this section, and add that you may need to configure the path (and that you likely want it to be "absolute"):

```suggestion
    <link rel="stylesheet" href="/path/to/simple-icons.min.css" type="text/css">
```

> <html>
>   <head>
>     <meta charset="utf-8">
>     <link rel="stylesheet" href="simple-icons.min.css" type="text/css">
>   </head>
>   <body>
>     <i class="simpleicons simpleicons-simpleicons simpleicons--color"></i>
>   </body>
> </html>

Reviewer: I would reduce this down to just the `<i>` tag and move it to the "Usage" section, as this is how you use it regardless of how you set it up.

```suggestion
    <i class="simpleicons simpleicons-simpleicons simpleicons--color"></i>
</html>
```

```

> ### Using module bundlers

Reviewer: Would prefer "Node Usage" here as that is what we use in the main repo. Also, the only relevant thing here is that you install it using NPM and use NodeJS to get it where you want. Whether you use bundlers or a script or something else entirely (including serving directly from `node_modules` :nauseated_face:) is up to the end user.

Author: Better just `NodeJS` to avoid confusion with `Usage` section?

Reviewer: Agreed! :+1:

```

> npm install simple-icons-font

Reviewer: You didn't change this, but to be inline with the main repo I think we should have

```suggestion
$ npm install simple-icons-font
```

```

After installation, the icon font can then be found in
`node_modules/simple-icons-font/font` along with stylesheets for a basic setup.

## Usage

The CSS files of simple-icons-font contains next classes:

- `simpleicons`: Adds the font to the container that will display the icon.
- `simpleicons--color`: Colorizes the icon with the main color of the brand.

> - `simpleicons-<ICON SLUG>`: Replace here `<ICON SLUG>` by with the

Reviewer: For consistency with the main repo, can we use square brackets here?

```suggestion
- `simpleicons-[ICON SLUG]`: Replace here `[ICON SLUG]` by with the
```

 correspondent slug for the icon. Their slugs correpond to filenames, which you
 can see listed [here][simple-icons--icons-dir-link]. 

## Status

[![Build Status][build-status-image]][build-status-link]
[![NPM version][npm-version-image]][npm-package-link]

[build-status-image]: https://img.shields.io/github/workflow/status/simple-icons/simple-icons-font/Verify/develop?logo=github
[build-status-link]: https://github.com/simple-icons/simple-icons-font/actions?query=workflow%3AVerify+branch%3Adevelop
[npm-version-image]: https://img.shields.io/npm/v/simple-icons-font?logo=npm
[npm-package-link]: https://www.npmjs.com/package/simple-icons-font
[simple-icons--icons-dir-link]: https://github.com/simple-icons/simple-icons/tree/develop/icons

> [npm-registry-tarball-link]: https://registry.npmjs.org/simple-icons-font/-/simple-icons-font-2.0.0.tgz

Reviewer: I would really be against having to update this for *every* release rather than every *major* release. Can we link to a page with all tarballs instead?

Author: Do you know how to get the directory where NPM serves all tarballs? :file_folder:
I have not found it anywhere...

Reviewer: Hmm :thinking: We could add [automated GitHub releases](https://github.com/simple-icons/simple-icons/blob/18cac9da6f091444bc391467067183ab2336b0af/.github/workflows/publish.yml#L28-L73) and link to the latest instead? E.g.

```
https://github.com/simple-icons/simple-icons/releases/latest
```

Currently links to:

```
https://github.com/simple-icons/simple-icons/releases/tag/4.2.0
```

And is automatically updated each release.

Author: Yes, would be the solution. I keep it like this and update when we add automated releases?

Reviewer: I will open a PR to set up automated releases later, we can see in which PR we change it later :+1:

Author: This will be unresolved until #54 merges. I think that we can wait for.

Reviewer: Sorry, didn't notice this comment until just after I merged this PR :sweat_smile: I will make sure to include this change in #54

Free SVG icon font for popular brands. See them all on one page at <a href="https://simpleicons.org">SimpleIcons.org</a>. Contributions, corrections & requests can be made on GitHub. Started by <a href="https://twitter.com/bathtype">Dan Leech</a>.</p>
</p>

## Setup

### From CDN

> The font can be served from a CDN such as [JSDelivr][jsdelivr-link] or [Unpkg][unkpkg-link]. Simply use the `simple-icons-font` npm package and specify a version in the URL like the following:

Reviewer: Small typo here breaks the link to Unpkg

```suggestion
The font can be served from a CDN such as [JSDelivr][jsdelivr-link] or [Unpkg][unpkg-link]. Simply use the `simple-icons-font` npm package and specify a version in the URL like the following:
```

> ```html
> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/simple-icons-font@v2/font/simple-icons.min.css" type="text/css">
> ```
> 
> ```html
> <link rel="stylesheet" href="https://unpkg.com/simple-icons-font@2/font/simple-icons.min.css" type="text/css">
> ```

Reviewer: Shall we put them together?

```suggestion
```html
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/simple-icons-font@v2/font/simple-icons.min.css" type="text/css">
<link rel="stylesheet" href="https://unpkg.com/simple-icons-font@2/font/simple-icons.min.css" type="text/css">
```
```

### NodeJS

The font is also available through our npm package. To install, simply run:

```
$ npm install simple-icons-font
```

> After installation, the icon and stylesheet font can then be found in `node_modules/simple-icons-font/font`. You can use your favorite bundling tools to include them into your project.

Reviewer: Was this a mistake in my suggestion? :fearful: 

```suggestion
After installation, the icon font and stylesheet can be found in `node_modules/simple-icons-font/font`. You can use your favorite bundling tool to include them into your project.
```

Reviewer: I'm 100% sure _"then"_ is unnecessary and 90% sure it is incorrect (though I'm not native in English either). This may be a bit easier to see if you simplify the sentence:

```diff
After installation, it can be found [...]
After installation, it can then be found [...]
```

### Manual download

You can also [download the latest version of the package][npm-registry-tarball-link] and copy the content of the `font` folder into your project. Then, reference the CSS file using a `link` tag in your HTML:

```html
<link rel="stylesheet" href="/path/to/simple-icons.min.css" type="text/css">
```

## Usage

Use any of the icons available in simple-icons by adding the following classes to a node in your HTML. Use the `simple-icons--color` class to apply the brand's color to the icon.

```html
<i class="simpleicons simpleicons-[ICON NAME]"></i>
<i class="simpleicons simpleicons-[ICON NAME] simple-icons--color"></i>
```

Where `[ICON NAME]` is replaced by the icon name, for example:

```html
<i class="simpleicons simpleicons-simpleicons"></i>
<i class="simpleicons simpleicons-simpleicons simple-icons--color"></i>
```

In this example we use the `<i>` tag, but any inline HTML tag should work as you expect.

> ### Example
> 
> ```html
> <i class="simpleicons simpleicons--color simpleicons-simpleicons"></i>
> ```

Reviewer: This example can be removed now, right?

```suggestion
```
